### ISSUE TITLE
Compile errors on macOS Ventura

### ISSUE THREAD

**Issue by dsully** [IndianBoy42/tree-sitter-just#15]
Getting a compile error when trying to use this via Neovim on macOS Ventura:

```
nvim-treesitter[just]: Error during compilation
src/scanner.cc:15:24: warning: default member initializer for non-static data member is a C++11 extension [-Wc++11-extensions]
  uint32_t prev_indent = 0;
                       ^
src/scanner.cc:47:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  auto curr = std::to_string(state->prev_indent);
  ^
src/scanner.cc:64:14: error: expected expression
    *state = {};
             ^
src/scanner.cc:88:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  auto advance = [lexer] { lexer->advance(lexer, false); };
  ^
src/scanner.cc:88:18: error: expected expression
  auto advance = [lexer] { lexer->advance(lexer, false); };
                 ^
src/scanner.cc:89:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  auto skip = [lexer] { lexer->advance(lexer, true); };
  ^
src/scanner.cc:89:15: error: expected expression
  auto skip = [lexer] { lexer->advance(lexer, true); };
              ^
src/scanner.cc:91:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  auto get_column = [lexer] { return lexer->get_column(lexer); };
  ^
src/scanner.cc:91:21: error: expected expression
  auto get_column = [lexer] { return lexer->get_column(lexer); };
                    ^
src/scanner.cc:128:5: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
    auto indent = get_column();
    ^
6 warnings and 4 errors generated.
```

**Comment by xbladesub** [comment]
Same

**Comment by dsully** [comment]
On macOS, setting this (and having `gcc` installed via Homebrew is work around:

```
require("nvim-treesitter.install").compilers = { "gcc", "clang" }
```

**Comment by on3iro** [comment]
I am still getting issues, even with the compiler installed:

```
nvim-treesitter[just]: Error during compilation
In file included from /opt/homebrew/Cellar/gcc@11/11.4.0/lib/gcc/11/gcc/aarch64-apple-darwin22/11/include/stdint.h:9,
                 from ./src/tree_sitter/parser.h:9,
                 from src/parser.c:1:
/opt/homebrew/Cellar/gcc@11/11.4.0/lib/gcc/11/gcc/aarch64-apple-darwin22/11/include-fixed/stdint.h:27:10: fatal error: sys/_types/_int8_t.h: No such file or d
irectory
   27 | #include <sys/_types/_int8_t.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [parser.o] Error 1
Press ENTER or type command to continue
```

Here's my tree-sitter-config-part:

```lua
			require("nvim-treesitter.install").compilers = { "gcc-11", "clang" }
			require("nvim-treesitter.parsers").get_parser_configs().just = {
				install_info = {
					url = "https://github.com/IndianBoy42/tree-sitter-just", -- local path or git repo
					files = { "src/parser.c", "src/scanner.cc" },
					branch = "main",
					use_makefile = true, -- this may be necessary on MacOS (try if you see compiler errors)
				},
				maintainers = { "@IndianBoy42" },
			}
```

**Comment by tgross35** [comment]
I removed C++11 features in 88404a4c23756426786581d09e6432bcdeec8b7c, this builds fine now
